jax/_src/lax/lax.py

```diff
@@ -2468,6 +2468,33 @@ def _named_call_translation_rule(c, axis_env, in_nodes, name_stack,
 
 xla.call_translations[named_call_p] = _named_call_translation_rule
 
+
+def named_call(
+    fun: Callable[..., Any],
+    *,
+    name: Optional[str] = None,
+) -> Callable[..., Any]:
+  """Wraps a function in a name_scope with the provided name.
```

Comment:
JAX does not have the concept of a "name scope" and we should not assume users are familiar with TF before using JAX. I think the important thing to call out is that named call is used to add names to staged jax programs.

Here's a concrete suggestion of a perhaps more useful doc string:

```
Adds a user specified name to a function when staging out JAX computations.

When staging out computations for just-in-time compilation to XLA (or other
backends such as TensorFlow) JAX runs your Python program but by default does
not preserve any of the function names or other metadata associated with it.
This can make debugging the staged out (and/or compiled) representation of your
program complicated because there is limited context information for each
operation being executed.

`named_call` tells JAX to stage the given function out as a subcomputation with
a specific name. When the staged out program is compiled with XLA these named
subcomputations are preserved and show up in debugging utilities like the
TensorFlow Profiler in TensorBoard. Names are also preserved when staging out
JAX programs to TensorFlow using :func:`experimental.jax2tf.convert`.
```

Reply:
I like this. Ideally we should have a documentation of how to interpret the name_stack that shows up in xprof.

Reply:
Thanks for the doc string, it looks very nice! 

> I like this. Ideally we should have a documentation of how to interpret the name_stack that shows up in xprof.

Should we add a screenshot and some explanations in this page? https://jax.readthedocs.io/en/latest/profiling.html?